src/mono/mono/component/mini-wasi-debugger.c

```diff
@@ -34,12 +34,10 @@ wasi_transport_recv (void *buf, int len)
 		if (res > 0)
 			total += res;
 		num_recv_calls++;
-		if (errno != 0)
-			return -1;
 		if ((res > 0 && total < len) || (res == -1 && num_recv_calls  < retry_receive_message)) {
 			// Wasmtime on Windows doesn't seem to be able to sleep for short periods like 1ms so we'll have to spinlock
 			long long start = timeInMilliseconds ();
-			while (timeInMilliseconds () < start + connection_wait_ms);
+			while (timeInMilliseconds () < start + (connection_wait_ms/1000));
```

Comment:
The naming is a bit odd now - is `connection_wait_ms` now measured in microseconds? Looks like it has to be for this comparison to be valid, but the name suggests milliseconds.

Reply:
If it is microseconds, could rename to `connection_wait_us` (I know the `us` suffix is a bit obscure though)